src/vs/workbench/contrib/terminal/browser/terminalMenus.ts

```diff
@@ -184,6 +185,39 @@ export function setupTerminalMenus(): void {
 					order: 3
 				}
 			},
+			{
+				id: MenuId.TerminalInstanceContext,
+				item: {
+					command: {
+						id: TerminalCommandId.RevealInFinder,
+						title: localize('reveal in finder', "Reveal in Finder"),
+					},
+					group: ContextMenuGroup.Navigation,
+					when: ContextKeyExpr.equals(TerminalContextKeyStrings.TerminalSettingShellIntegrationEnabled, true)
+				},
+			},
+			{
+				id: MenuId.TerminalInstanceContext,
+				item: {
+					command: {
+						id: TerminalCommandId.RevealInExplorer,
+						title: localize('reveal in explorer', "Reveal in Explorer"),
+					},
+					group: ContextMenuGroup.Navigation,
+					when: ContextKeyExpr.equals(TerminalContextKeyStrings.TerminalSettingShellIntegrationEnabled, true)
+				}
+			},
+			{
+				id: MenuId.TerminalInstanceContext,
+				item: {
+					command: {
+						id: TerminalCommandId.OpenInExternalTerminal,
+						title: localize('open in external terminal', "Open in External Terminal"),
+					},
+					group: ContextMenuGroup.Navigation,
+					when: ContextKeyExpr.equals(TerminalContextKeyStrings.TerminalSettingShellIntegrationEnabled, true)
```

Comment:
These when clauses would be better off as preconditions on the command itself, not the context menu entry.